pkg/sinks/sink.go

```diff
@@ -32,6 +35,15 @@ func (u *SinkProcessor) Start(ctx context.Context) error {
 	var reader isb.BufferReader
 	var err error
 	fromBufferName := u.VertexInstance.Vertex.GetFromBuffers()[0].Name
+
+	// watermark variables no-op initialization
+	var fetchWatermark fetch.Fetcher = generic.NewNoOpWMProgressor()
+	// publishWatermark is a map representing a progressor per edge, we are initializing them to a no-op progressor
+	publishWatermark := make(map[string]publish.Publisher)
+	for _, buffer := range u.VertexInstance.Vertex.GetToBuffers() {
```

Comment:
Be aware `GetToBuffers()` returns a particular `sink buffer`, make sure this is what we want.

Reply:
Yes, we are looking for the sink buffer for checking this